app/controllers/jobseekers/job_applications/qualifications_controller.rb

```diff
@@ -0,0 +1,98 @@
+class Jobseekers::JobApplications::QualificationsController < Jobseekers::BaseController
+  helper_method :back_path, :form, :job_application, :qualification
+
+  def submit_category
+    if params[:commit] == t("buttons.cancel")
+      redirect_to back_path
+    elsif form.valid?
+      redirect_to new_jobseekers_job_application_qualification_path(qualification_params)
+    else
+      render :select_category
+    end
+  end
+
+  def create
+    if params[:commit] == t("buttons.cancel")
+      redirect_to back_path
+    elsif form.valid?
+      job_application.qualifications.create(qualification_params)
+      redirect_to back_path
+    else
+      render :new
+    end
+  end
+
+  def update
+    if params[:commit] == t("buttons.cancel")
+      redirect_to back_path
+    elsif form.valid?
+      qualification.update(qualification_params)
+      redirect_to back_path
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    qualification.destroy
+    redirect_to back_path, success: t(".success")
+  end
+
+  private
+
+  def form
+    @form ||= action_name == "select_category" ? form_class.new : form_class.new(qualification_params)
```

Comment:
Should this use the `form_attributes` method?

```ruby
@form ||= form_class.new(form_attributes)
```

and update the case...when to be 
```ruby
when "new", "select_category"
  {}
```